src/sys/linux/pipe.rs

```diff
@@ -2,4 +2,63 @@
 #![allow(unused)]
 
 pub enum read2 { }
-pub enum AnonPipe { }
+use sync::atomic::{AtomicBool, Ordering};
+use cmp;
+use io;
+use linux;
+use mem;
+use ptr;
+use sys::{cvt, errno};
+use sys::fd::FileDesc;
+
+////////////////////////////////////////////////////////////////////////////////
+// Anonymous pipes
+////////////////////////////////////////////////////////////////////////////////
+
+pub struct AnonPipe(FileDesc);
+
+pub fn anon_pipe() -> io::Result<(AnonPipe, AnonPipe)> {
+    let mut fds = [0; 2];
+
+    // Unfortunately the only known way right now to create atomically set the
```

Comment:
No. `weak!` tries to dynamically bind to symbols provided by the libc, which we do not have. Try to call the syscall directly, it'll return `-ENOSYS` if it doesn't exist on the host platform.